frontend/app/player/MessageDistributor/managers/AssistManager.ts

```diff
@@ -239,12 +248,57 @@ export default class AssistManager {
         this.toggleRemoteControl(false)
       })
       socket.on('call_end', this.onRemoteCallEnd)
+      socket.on('recording_accepted', () => {
+        this.toggleRecording(true)
+      })
+      socket.on('recording_rejected', () => {
+        this.toggleRecording(false)
+      })
 
       document.addEventListener('visibilitychange', this.onVisChange)
 
     })
   }
 
+  /* ==== Recording the session ==== */
+
+  public requestRecording = () => {
+    const recordingState = getState().recordingState
+    if (!this.socket || recordingState === SessionRecordingStatus.Requesting) return;
+
+    update({ recordingState: SessionRecordingStatus.Requesting })
+    this.socket.emit("request_recording", JSON.stringify({
+      ...this.session.agentInfo,
+      query: document.location.search,
+    }))
+  }
+
+  public stopRecording = () => {
+    const recordingState = getState().recordingState
+    if (!this.socket || recordingState === SessionRecordingStatus.Off) return;
+
+    this.socket.emit("stop_recording")
+    this.toggleRecording(false)
+  }
+
+  private get borderStyle() {
+    const { recordingState, remoteControl } = getState()
+
+    const isRecordingActive = recordingState === SessionRecordingStatus.Recording
+    const isControlActive = remoteControl === RemoteControlStatus.Enabled
+    const baseBorder = '2px dashed'
+    // recording gets priority here
+    if (isRecordingActive) return { border: `${baseBorder} red`}
+    if (isControlActive) return { border: `${baseBorder} blue`}
```

Comment:
since there is a small change request anyway: 
having `const baseBorder` increases amount of symbols and calculations, better put string directly